src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -325,9 +325,148 @@ module.exports = class LevelPlane extends Array {
+        if (neighbors.east.block.blockType !== "" && neighbors.east.block.blockType !== "pistonArm") {
+          let offset = [1,0];
+          let blocksPositions = this.getBlocksToPush(neighborPosition[2], offset);
+          this.pushBlocks(blocksPositions, offset);
+        } else if (neighbors.east.block.blockType === "") {
+          let armBlock = new LevelBlock("pistonArm");
+          this.setBlockAt(neighborPosition[2], armBlock);
+        }
+        break;
+      }
+      case "pistonLeft": {
+        if (neighbors.west.block.blockType !== "" && neighbors.west.block.blockType !== "pistonArm") {
+          let offset = [-1,0];
+          let blocksPositions = this.getBlocksToPush(neighborPosition[3], offset);
+          this.pushBlocks(blocksPositions, offset);
+        } else if (neighbors.west.block.blockType === "") {
+          let armBlock = new LevelBlock("pistonArm");
+          this.setBlockAt(neighborPosition[3], armBlock);
+        }
+        break;
+      }
+    }
+  }
+
+  deactivatePiston(position) {
+    let neighborPosition = this.getOrthogonalPositions(position);
+    let emptyBlock = new LevelBlock("");
+
+    switch (this[this.coordinatesToIndex(position)].blockType) {
+      case "pistonUp": {
+        if (this[this.coordinatesToIndex(neighborPosition[0])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[0], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonDown": {
+        if (this[this.coordinatesToIndex(neighborPosition[1])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[1], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonRight": {
+        if (this[this.coordinatesToIndex(neighborPosition[2])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[2], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonLeft": {
+        if (this[this.coordinatesToIndex(neighborPosition[3])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[3], emptyBlock, false, true);
+        }
+        break;
+      }
+    }
+  }
+
+  pushBlocks(blocksPositions, offset) {
+    let armBlock = new LevelBlock("pistonArm");
+    // Need to check if the last index is up against the edge of the map, and break out if that's the case.
+    if (!this.inBounds([blocksPositions[blocksPositions.length - 1][0] + offset[0], blocksPositions[blocksPositions.length - 1][1] + offset[1]])) {
+      // Apparently me might NOT want this check? Or might? Design undecided.
+      //break;
+    }
+
+    for (let i = blocksPositions.length - 1; i >= 0; --i) {
+      let destination = [blocksPositions[i][0] + offset[0], blocksPositions[i][1] + offset[1]];
+      this.setBlockAt(destination, this.getBlockAt(blocksPositions[i]));
+      if (i === 0) {
+        this.setBlockAt(blocksPositions[i], armBlock);
+      }
+    }
+  }
+
+  getBlocksToPush(position, offset) {
```

Comment:
Maybe worth matching the signature of `getBlockAt` where `offsetX` and `offsetY` are passed in separately?

```js
getBlockAt(position, offsetX = 0, offsetY = 0) { ... }
```

It's not clear to me that either one is better than the other, but my vote would be for them to be consistent 😄.

Reply:
That's a good point. Will change 👍